lib/config.js

```diff
@@ -422,9 +413,33 @@ function normalize (opts, logger) {
   normalizeIgnoreOptions(opts)
   normalizeSanitizeFieldNames(opts)
   normalizeCloudProvider(opts, logger)
+  normalizeTransactionSampleRate(opts, logger)
   truncateOptions(opts)
 }
 
+// transactionSampleRate is specified to be:
+// - in the range [0,1]
+// - rounded to 4 decimal places of precision (e.g. 0.0001, 0.5678, 0.9999)
+// - with the special case that a value in the range (0, 0.0001] should be
+//   rounded to 0.0001 -- to avoid a small value being rounded to zero.
+//
+// https://github.com/elastic/apm/blob/master/specs/agents/tracing-sampling.md
+function normalizeTransactionSampleRate (opts, logger) {
+  if ('transactionSampleRate' in opts) {
+    // The value was already run through `Number(...)` in `normalizeNumbers`.
+    const rate = opts.transactionSampleRate
+    if (isNaN(rate) || rate < 0 || rate > 1) {
+      opts.transactionSampleRate = DEFAULTS.transactionSampleRate
```

Comment:
I see this conditional performs a similar validation to the number that was done previously by `numberBetweenZeroAndOne`.  Also, since (per the comment) we've already nun the value through `Number`,  `isNaN` won't "false position" on numbers as strings.  👍